evil-states.el

```diff
@@ -350,17 +350,13 @@ otherwise exit Visual state."
 
 (defun evil-visual-update-x-selection (&optional buffer)
   "Update the X selection with the current visual region."
-  (let ((buf (or buffer (current-buffer))))
-    (when (buffer-live-p buf)
```

Comment:
Why did you remove the `buffer-live-p` check?

Reply:
Because I based the code on the version from 2012 that didn't have this and completely ignored this part of the diff. Thanks for noticing, I fixed it in a new commit (and added a test).